sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchemaSuite.scala

```diff
@@ -192,4 +192,27 @@ class CSVInferSchemaSuite extends SparkFunSuite with SQLHelper {
     Seq("en-US").foreach(checkDecimalInfer(_, StringType))
     Seq("ko-KR", "ru-RU", "de-DE").foreach(checkDecimalInfer(_, DecimalType(7, 0)))
   }
+
+  test("SPARK-34953 - DateType should be inferred when user defined format are provided") {
```

Comment:
The common convention is: `SPARK-XXXXX: ...`
```suggestion
  test("SPARK-34953: DateType should be inferred when user defined format are provided") {
```

Reply:
done.